aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -330,6 +330,75 @@ void linalg_eigh_kernel(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos
       });
 }
 
+/*
+  The geqrf function computes QR decomposition of matrices stored in `input`.
+  However, rather than producing a Q matrix directly, it produces a sequence of
+  elementary reflectors which may later be composed to construct Q - for example
+  with the orgqr or ormqr functions.
+
+  Args:
+  * `input` - [in] Input tensor for QR decomposition
+             [out] QR decomposition result which contains:
+              i)  The elements of R, on and above the diagonal.
+              ii) Directions of the reflectors implicitly defining Q.
+             Tensor with the directions of the elementary reflectors below the diagonal,
+              it will be overwritten with the result
+  * `tau` - [out] Tensor which will contain the magnitudes of the reflectors
+            implicitly defining Q.
+  * `m` - The number of rows of `input` to consider
+  * `n` - The number of columns of `input` to consider (actual sizes of `input` could be larger)
+
+  For further details, please see the LAPACK documentation for GEQRF.
+*/
+template <typename scalar_t>
+static void apply_geqrf(const Tensor& input, const Tensor& tau, int64_t m, int64_t n) {
+#ifndef USE_LAPACK
+  AT_ERROR("geqrf: LAPACK library not found in compilation");
```

Comment:
TORCH_CHECK and double check this error message for consistency with other LAPACK not found errors, I think we often say something diferent?